blocks/live-now/live-now.js

```diff
@@ -0,0 +1,221 @@
+// set session cookie
+function setSessionCookie(name, value) {
+  document.cookie = `${name}=${value}; path=/`;
+}
+
+// get existing session cookie
+function getSessionCookie(name) {
+  const value = `; ${document.cookie}`;
+  const parts = value.split(`; ${name}=`);
+  if (parts.length === 2) return parts.pop().split(';').shift();
+  return null;
+}
+
+// check cookie for event guid
+function hasModalBeenDisplayed(cookieName, guid) {
+  const eventCookie = getSessionCookie(cookieName);
+  if (!eventCookie) return false; // no cookie found, modal for event hasn't been seen
+
+  const eventList = eventCookie.split(','); // split stored GUIDs into an array
+  return eventList.includes(guid); // check if the event GUID is in the array
+}
+
+// record event guid if modal displays
+function addGuidToCookie(cookieName, guid) {
+  let eventCookie = getSessionCookie(cookieName);
+  if (eventCookie) {
+    eventCookie += `,${guid}`; // append guid to cookie
+  } else {
+    eventCookie = guid; // first guid/cookie
+  }
+  setSessionCookie(cookieName, eventCookie); // create or update the cookie
+}
+
+// retrieve event data from API
+async function queryEventData(dateTime) {
+  const endpoint = 'https://webapi.worldbank.org/aemsite/wblive/global/search';
+  const subkey = 'a02440fa123c4740a83ed288591eafe4';
```

Comment:
Maybe as a follow up, we might want to give the author an option to author this URL and subscription key,